src/io/contreaddata.c

```diff
@@ -61,6 +64,9 @@ static void __osPackReadData(void) {
     readformat.stick_y = -1;
 
     for (i = 0; i < __osMaxControllers; i++) {
+        if ((__osControllerMask & (1 << i)) == 0) {
+            continue;
+        }
```

Comment:
I have two issues with this:
- I'm pretty sure as-written, this will query the physical joybus channels in order even if the current controller mask has gaps. This is because the PIF packet being built doesn't instruct the PIF to skip this channel, it just writes nothing. It might be as easy as adding `*ptr++ = 0x00;` to instruct the PIF to skip a channel.
- This will eventually break iQue compatibility, iQue needs to query all channels in a very specific way. Specifically each channel needs to be 8 bytes starting with FF, PIF channel skipping is not supported. This feature may need to be ignored altogether for iQue.